data_manager.rb

```diff
@@ -0,0 +1,48 @@
+class PeopleManager
+  attr_reader :people
+
+  def initialize(people)
+    @people = people
+  end
+
+  def list_people
+    @people.map do |person|
+      tag = person.is_a?(Student) ? 'Student' : 'Teacher'
+      puts "[#{tag}] ID: #{person.id} Name: #{person.name}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    puts 'Are you adding a student (1) or a teacher (2)?'
+    print '[Input a number]: '
+    person_instance = gets.chomp.to_i
+    case person_instance
+    when 1, 2
+      print 'Age: '
+      age = gets.chomp
+      print 'Name: '
+      name = gets.chomp
+      person_instance == 1 ? create_student(age, name) : create_teacher(age, name)
```

Comment:
Just as I have mentioned in the previous comment, the same thing applies here😉. Kindly ensure every input request and input validation is done in the `main.rb` file😉.